### PR TITLE
Remove chalk-base

### PR DESCRIPTION
**Pull request by jackh726**

### CODE REVIEW

File: chalk-engine/src/lib.rs [hunk 1]
```diff
@@ -285,3 +282,50 @@ chalk_ir::copy_fold!(TimeStamp);
 
 chalk_ir::const_visit!(TableIndex);
 chalk_ir::const_visit!(TimeStamp);
+
+#[macro_export]
+macro_rules! index_struct {
```
Comment (Mcat12) [inline chalk-engine/src/lib.rs:287]: Good idea, moving to chalk-engine!

File: chalk-ir/src/lib.rs [hunk 2]
```diff
@@ -1,13 +1,15 @@
 #![deny(rust_2018_idioms)]
 
+#[macro_use]
+extern crate lazy_static;
```
Comment (Mcat12) [inline chalk-ir/src/lib.rs:4]: Can we directly import `lazy_static::lazy_static`, or is it still idiomatic to `macro_use` it (compared to when we directly import the debug macros)?

Comment (jackh726) [reply chalk-ir/src/lib.rs:4]: Good catch! I'll change this.

**Comment by jackh726** [comment]
Thanks @Mcat12 for the review. I also snuck in the two test fixes from my boo-boo in https://github.com/rust-lang/chalk/pull/477.

Once CI is green, I'll go ahead and merge.

**Pull request merged**